src/Symfony/Component/String/Slugger/AsciiSlugger.php

```diff
@@ -119,9 +121,23 @@ public function slug(string $string, string $separator = '-', string $locale = n
 
         $unicodeString = (new UnicodeString($string))->ascii($transliterator);
 
-        if (\is_array($this->symbolsMap) && isset($this->symbolsMap[$locale])) {
-            foreach ($this->symbolsMap[$locale] as $char => $replace) {
-                $unicodeString = $unicodeString->replace($char, ' '.$replace.' ');
+        if (\is_array($this->symbolsMap)) {
+            $map = null;
+            // if we have a symbols map for the requested locale, use it
```

Comment:
no need for the comment nor for the initialization

Reply:
I think the initialization is needed, in case neither $locale nor $parent are present in the $symbolsMap array.